src/components/common/ColonyMembers/MembersFilter/MembersFilter.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useRef } from 'react';
+import { FormikProps } from 'formik';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import Button from '~shared/Button';
+import { Form, Select } from '~shared/Fields';
+
+import styles from './MembersFilter.css';
+
+const displayName = 'common.ColonyMembers.MembersFilter';
+
+const MSG = defineMessages({
+  filter: {
+    id: `${displayName}.filter`,
+    defaultMessage: 'Filters',
+  },
+  reset: {
+    id: `${displayName}.reset`,
+    defaultMessage: 'Reset',
+  },
+  allMembers: {
+    id: `${displayName}.allMembers`,
+    defaultMessage: 'All members',
+  },
+  any: {
+    id: `${displayName}.allMembers`,
+    defaultMessage: 'Any',
+  },
+  contributors: {
+    id: `${displayName}.contributors`,
+    defaultMessage: 'Contributors',
+  },
+  watchers: {
+    id: `${displayName}.watchers`,
+    defaultMessage: 'Watchers',
+  },
+  verified: {
+    id: `${displayName}.verified`,
+    defaultMessage: 'Verified',
+  },
+  unverified: {
+    id: `${displayName}.unverified`,
+    defaultMessage: 'Unverified',
+  },
+  banned: {
+    id: `${displayName}.banned`,
+    defaultMessage: 'Banned',
+  },
+  notBanned: {
+    id: `${displayName}.notBanned`,
+    defaultMessage: 'Not banned',
+  },
+  memberType: {
+    id: `${displayName}.memberType`,
+    defaultMessage: 'Member type',
+  },
+  bannedStatus: {
+    id: `${displayName}.bannedStatus`,
+    defaultMessage: 'Banned status',
+  },
+  verificationType: {
+    id: `${displayName}.verificationType`,
+    defaultMessage: 'Verification type',
+  },
+});
+
+export enum MemberType {
+  ALL = 'all',
+  CONTRIBUTORS = 'contributors',
+  WATCHERS = 'watchers',
+}
+
+export enum VerificationType {
+  ALL = 'all',
+  VERIFIED = 'verified',
+  UNVERIFIED = 'unverified',
+}
+
+export enum BannedStatus {
+  ALL = 'all',
+  BANNED = 'banned',
+  NOT_BANNED = 'not_banned',
+}
+
+export interface FormValues {
+  memberType: MemberType;
+  verificationType: VerificationType;
+  bannedStatus: BannedStatus;
+}
+
+const memberTypes = [
+  { label: MSG.allMembers, value: MemberType.ALL },
+  { label: MSG.contributors, value: MemberType.CONTRIBUTORS },
+  { label: MSG.watchers, value: MemberType.WATCHERS },
+];
+
+const verificationTypes = [
+  { label: MSG.any, value: VerificationType.ALL },
+  { label: MSG.verified, value: VerificationType.VERIFIED },
+  { label: MSG.unverified, value: VerificationType.UNVERIFIED },
+];
+
+const bannedStatuses = [
```

Comment:
These arrays and enums could be moved into their own file, along with the encapsulated filters logic (see  comment below)